LanguageFeatures/Patterns/shared_case_scope_A02_t03.dart

```diff
@@ -44,7 +44,7 @@ String test1(num? x) {
   switch (x) {
     case final a when a is int:
```

Comment:
We can make sure that `a` is promoted to `num` in every branch (and may or may not be promoted further) by including an `a is num`. So:
```suggestion
    case final a when a is num && a is int:
```
The promotion chains would then be `[int, num, num?]` and `[num, num?]`, and the intersection is `[num, num?]` and hence `a` has type `num`.